src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -17903,5 +17925,1356 @@ public class XAttribute : System.Attribute
+                {
+                    IEnumerator IEnumerable.GetEnumerator() => null;
+                    IEnumerator<long> IEnumerable<long>.GetEnumerator() => null;
+                    IEnumerator<int> IEnumerable<int>.GetEnumerator() => null;
+
+                    public void Add(int i) { System.Console.Write("RAN"); }
+                    public void Add(long s) => throw null;
+                }
+                """;
+
+            CreateCompilation(source).VerifyEmitDiagnostics(
+                // (4,16): error CS9174: Cannot initialize type 'Collection' with a collection expression because the type is not constructible.
+                // Collection c = ["hi"];
+                Diagnostic(ErrorCode.ERR_CollectionExpressionTargetTypeNotConstructible, @"[""hi""]").WithArguments("Collection").WithLocation(4, 16)
+                );
+        }
+
+        [Theory]
+        [InlineData("class")]
+        [InlineData("struct")]
+        public void GenericIEnumerable_SingleInvalid(string typeKind)
+        {
+            string source = $$"""
+                using System.Collections;
+                using System.Collections.Generic;
+
+                Collection c = ["hi", null];
+
+                {{typeKind}} Collection : IEnumerable<int>
+                {
+                    IEnumerator IEnumerable.GetEnumerator() => null;
+                    IEnumerator<int> IEnumerable<int>.GetEnumerator() => null;
+
+                    public void Add(int i) { System.Console.Write("RAN"); }
+                }
+                """;
+
+            CreateCompilation(source).VerifyEmitDiagnostics(
+                // (4,17): error CS0029: Cannot implicitly convert type 'string' to 'int'
+                // Collection c = ["hi", null];
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, @"""hi""").WithArguments("string", "int").WithLocation(4, 17),
+                // (4,23): error CS0037: Cannot convert null to 'int' because it is a non-nullable value type
+                // Collection c = ["hi", null];
+                Diagnostic(ErrorCode.ERR_ValueCantBeNull, "null").WithArguments("int").WithLocation(4, 23)
+                );
+        }
+
+        [Fact]
+        public void GenericIEnumerable_SingleInvalid_TypeParameter()
+        {
+            string source = """
+                using System.Collections;
+                using System.Collections.Generic;
+
+                class C
+                {
+                    T M<T>() where T : ICollection
+                    {
+                        return ["hi", null];
+                    }
+                }
+
+                interface ICollection : IEnumerable<int>
+                {
+                    IEnumerator IEnumerable.GetEnumerator() => null;
+                    IEnumerator<int> IEnumerable<int>.GetEnumerator() => null;
+
+                    public void Add(int i) { System.Console.Write("RAN"); }
```

Comment:
Consider changing these to interface methods with no implementation. I don't think there should be any impact on binding, but it seems unnecessary to include that.